atlite/convert.py

```diff
@@ -128,22 +129,56 @@ def convert_and_aggregate(
             res = da.sum("time", keep_attrs=True)
             return maybe_progressbar(res, show_progress, **dask_kwargs)
 
+    if matrix is not None:
+
+        if shapes is not None:
+            raise ValueError(
+                "Passing matrix and shapes is ambiguous. Pass " "only one of them."
+            )
+
+        if isinstance(matrix, xr.DataArray):
+
+            if {"x", "y"}.issubset(matrix.dims):
+                matrix = matrix.stack(spatial=["y", "x"])
+
+            # ensure correct alignment of spatial axis
+            coords = matrix.indexes.get(matrix.dims[1]).to_frame(index=False)
+            if not np.array_equal(coords[["x", "y"]], cutout.grid[["x", "y"]]):
+                raise ValueError(
+                    "Matrix spatial coordinates not aligned with cutout spatial "
+                    "coordinates."
+                )
+
+            if index is None:
+                index = matrix
+
+        if not matrix.ndim == 2:
+            raise ValueError("Matrix not 2-dimensional.")
```

Comment:
Just to emphasize my point from above: This error message is very confusing to people leveraging `matrix` with a 3D `xr.DataArray`.